client/gutenberg/editor/controller.js

```diff
@@ -142,6 +143,30 @@ export const redirect = ( { store: { getState } }, next ) => {
 	return page.redirect( `/post/${ getSelectedSiteSlug( state ) }` );
 };
 
+function getPressThisContent( query ) {
+	const { text, url, title, image, embed } = query;
+	const caption = !! text ? '' : '<figcaption><a href="${ url }">${ title }</a></figcaption>';
+	let content = '';
+
+	if ( isValidUrl( embed ) ) {
+		//TODO: this turns into an wp:html block
+		content = content + `<!-- wp:embed --><p>${ embed }</p><!-- /wp:embed -->`;
```

Comment:
If we move the `getPressThisContent` call after `initGutenberg` has been called ([#](https://github.com/Automattic/wp-calypso/pull/29878/files#diff-f319ed205677209241d77fdb7197d1a3R195)), the Core blocks will be already registered, so we can use `serialize` from `@wordpress/blocks` instead of defining the HTML output.

```js
import { createBlock, serialize } from '@wordpress/blocks';

const embedBlock = createBlock( 'core/embed', { url: embed }  );
content += serialize( embedBlock );

const imageBlock = createBlock( 'core/image', { url: image, caption }  );
content += serialize( imageBlock );

const quoteBlock = createBlock( 'core/quote', { value: text, citation: '<a href="${ url }">${ title }</a>' }  );
content += serialize( quoteBlock );
```